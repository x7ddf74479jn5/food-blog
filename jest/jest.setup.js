// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
