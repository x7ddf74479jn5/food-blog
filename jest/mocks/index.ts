export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
