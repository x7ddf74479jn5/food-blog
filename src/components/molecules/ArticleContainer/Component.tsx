import ArticleList from "@/components/molecules/ArticleList";
import Pagination from "@/components/molecules/Pagination";
import useGetArticleListQuery from "@/hooks/useGetArticleListQuery";

export const Component: React.VFC = () => {
  const { data, error, size, setSize, isValidating } = useGetArticleListQuery({
    perPage: 2,
    options: { suspense: true },
  });
  // const ArticleList = dynamic(() => import("@/components/molecules/ArticleList"),{ssr:false});
  const handleOnClick = () => {
    setSize(size + 1);
  };

  const articles = data ? data.map((r) => (r ? r.contents : [])).flat() : [];
  const totalCount = data ? data[0]?.totalCount : 0;
  const hasNextPage = totalCount ? articles.length !== totalCount : false;

  if (error) return <div className="flex justify-center mt-16">エラーが発生しました。</div>;

  if (articles.length === 0) return <div className="flex justify-center mt-16">レシピが見つかりませんでした。</div>;

  return (
    <>
      <ArticleList articles={articles} />
      <Pagination hasNextPage={hasNextPage} isValidating={isValidating} onClick={handleOnClick} />
    </>
  );
};

export default Component;
