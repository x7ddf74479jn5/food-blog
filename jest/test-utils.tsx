import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
