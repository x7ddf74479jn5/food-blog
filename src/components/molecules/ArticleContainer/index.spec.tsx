import { render, screen } from "jest/test-utils";
// import { server } from "mocks/msw/server";
// import { Suspense } from "react";
import renderer from "react-test-renderer";

// import { HttpErrorBoundary } from "@/components/atoms/error";
// import { ArticleSkeltonList } from "@/components/molecules/ArticleSkeltonList";
import { ArticleContainer } from ".";
import Component from "./Component";

// beforeAll(() => server.listen());
// afterAll(() => server.close());

// jest.mock("@/hooks/useGetArticleListQuery", () => {
//   const useGetArticleListQuery = jest.fn().mockReturnValue({});
// });

// const mockRouter = { query: { q: "copy" } };

describe("components/molecules/ArticleContainer", () => {
  it("should ", async () => {
    // <HttpErrorBoundary>
    //   <Suspense fallback={<ArticleSkeltonList />}>
    //     <Component />
    //   </Suspense>
    // </HttpErrorBoundary>;

    await render(<ArticleContainer />);
    // await Component;
    await Component;

    // await render(
    //   <HttpErrorBoundary>
    //     <Suspense fallback={<ArticleSkeltonList />}>
    //       <Component />
    //     </Suspense>
    //   </HttpErrorBoundary>
    // );
    // await Component;

    screen.debug();
  });

  it("snapshot", async () => {
    await Component;

    // const test = withMockedRouter(
    //   mockRouter,
    //   <HttpErrorBoundary>
    //     <Suspense fallback={<ArticleSkeltonList />}>
    //       <Component />
    //     </Suspense>
    //   </HttpErrorBoundary>
    // );

    // const tree = await renderer.create(test).toJSON();
    // const tree = await renderer.create(<Component />).toJSON();
    const tree = await renderer.create(<ArticleContainer />).toJSON();
    // await Component;
    expect(tree).toMatchSnapshot();
  });
});
