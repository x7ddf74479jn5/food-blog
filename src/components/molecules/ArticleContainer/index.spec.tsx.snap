// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/ArticleContainer snapshot 1`] = `
<div
  className="flex flex-col gap-4 justify-center items-center"
  role="alert"
>
  <p>
    Something went wrong:
  </p>
  <pre>
    Cannot read property 'query' of null
  </pre>
  <button
    onClick={[Function]}
  >
    Try again
  </button>
</div>
`;
